modules/ROOT/pages/dataweave-cookbook-extract-data.adoc

```diff
@@ -117,24 +117,30 @@ length (`.^contentLength`), encoding (`.^encoding`), MIME type (`.^mimeType`),
 media type (`.^mediaType`), raw  (`.^raw`), and custom (`.^myCustomMetadata`)
 metadata. For details, see <<metadata_selector>>.
 
-== Valid Keys
+== Key Syntax
 
-To be valid, a key name in a DataWeave object must start with a letter that is
+To validate, a key name in a DataWeave object must start with a letter that is
```

Comment:
Does it make sense to say that any key name is valid but the syntax to use them changes depending if they are valid identifiers (https://docs.mulesoft.com/mule-runtime/4.3/dataweave-language-introduction#valid-identifiers) or not? Because something that is not mentioned here is that key names cannot be reserved keywords(to be considered valid).
I'm not sure if by reading this it is completely clear that anything can be a valid key, the only thing that changes is when it is necessary to use quotes.

Reply:
Good point. Tested with a reserved keyword (`type`). `{ type : "something" }` throws an error (`Invalid input "ype ", expected Namespace (line 4, column 4):`), but `{ "type" : "something" }` works fine.

Reply:
I'll add a link to https://docs.mulesoft.com/mule-runtime/4.3/dataweave-language-introduction#valid-identifiers

Reply:
done, maybe we can create a Jira to use an include on any shared content.

Reply:
As a developer who learned quite some languages "Valid" or "Invalid" Keywords imples, that Invalid Keywords are the one I in no way can use. Now per https://docs.mulesoft.com/mule-runtime/4.3/dataweave-language-introduction#valid-identifiers `123456` would be considered an invalid keyword although I can actually use `123456` like here: `{ "123456" : "something" }` which now contradicts the implications the words "Valid" and "Invalid" have.

Reply:
What we need to clarify is that DataWeave accepts any kind of key, the difference being that if the key is a _valid identifier_, no quotes are needed surrounding it.

Reply:
Ok, we have discussed with DW dev and Eugen. We'll clarify a distinction bt identifiers and keys.